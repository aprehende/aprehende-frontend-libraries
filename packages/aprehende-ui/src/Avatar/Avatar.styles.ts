import styled from '@emotion/styled';

export const AvatarStyled = styled.div<{ picture?: string }>`
  width: 2rem;
  height: 2rem;
  padding: 0.5rem;
  font-size: 1.3rem;
  border-radius: 50%;
  background-size: cover;
  border: 1px solid black;
  background-position: center;
  background-repeat: no-repeat;
  ${({ picture }) => picture && `background-image: url(${picture});`}
`;

export const AvatarTextStyled = styled.span`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
`;
