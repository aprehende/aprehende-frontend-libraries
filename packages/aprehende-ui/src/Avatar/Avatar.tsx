import { useMemo } from 'react';

import { AvatarStyled, AvatarTextStyled } from './Avatar.styles';
import { useAvatar } from './hooks';

interface AvatarProps {
  name: string;
  picture?: string;
}

const Avatar: React.FC<AvatarProps> = ({ name, picture }) => {
  const { getInitials } = useAvatar();

  const initials = useMemo(() => getInitials(name), [getInitials, name]);

  return (
    <AvatarStyled picture={picture}>
      {!picture && <AvatarTextStyled>{initials}</AvatarTextStyled>}
    </AvatarStyled>
  );
};

export default Avatar;
