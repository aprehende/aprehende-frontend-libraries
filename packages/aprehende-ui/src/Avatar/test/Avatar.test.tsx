import { render } from '@testing-library/react';
import { describe, it } from 'vitest';

import Avatar from '../Avatar';

describe('Avatar', () => {
  it('should render the initials', () => {
    const { getByText } = render(<Avatar name="Jordan Ruben" />);

    expect(getByText('JR')).toBeInTheDocument();
  });

  it('snapshot with picture and name', () => {
    const { container } = render(
      <Avatar
        name="Jordan Ruben"
        picture="https://cdn4.buysellads.net/uu/1/134955/1685040267-designdotdev5.jpg"
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('snapshot with name', () => {
    const { container } = render(<Avatar name="Jordan Ruben" />);

    expect(container).toMatchSnapshot();
  });

  it('should return error if name is not provided', () => {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    expect(() => render(<Avatar />)).toThrowError();
  });
});
