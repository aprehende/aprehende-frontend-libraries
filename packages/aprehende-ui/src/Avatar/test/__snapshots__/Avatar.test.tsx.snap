// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar > snapshot with name 1`] = `
<div>
  <div
    class="css-1dw3puw"
  >
    <span
      class="css-16rvg39"
    >
      JR
    </span>
  </div>
</div>
`;

exports[`Avatar > snapshot with picture 1`] = `
<div>
  <div
    class="css-jbmq27"
  />
</div>
`;

exports[`Avatar > snapshot with picture and name 1`] = `
<div>
  <div
    class="css-jbmq27"
  />
</div>
`;
